libs/feature/search/src/lib/results-table/results-table.component.spec.ts

```diff
@@ -0,0 +1,209 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { DATASET_RECORDS } from '@geonetwork-ui/common/fixtures'
+import { ResultsTableComponent } from './results-table.component'
+import { CatalogRecord } from '@geonetwork-ui/common/domain/model/record'
+import { By } from '@angular/platform-browser'
+import { BehaviorSubject, firstValueFrom } from 'rxjs'
+import { SearchFacade } from '../state/search.facade'
+import { SearchService } from '../utils/service/search.service'
+import { SelectionService } from '@geonetwork-ui/api/repository'
+import { TranslateModule } from '@ngx-translate/core'
+
+class SearchFacadeMock {
+  results$ = new BehaviorSubject(DATASET_RECORDS)
+  resultsHits$ = new BehaviorSubject(1000)
+  setConfigRequestFields = jest.fn(() => this)
+  setSortBy = jest.fn(() => this)
+  sortBy$ = new BehaviorSubject<any>(['asc', 'updateDate'])
+}
+class SearchServiceMock {
+  setPage = jest.fn()
+  setSortBy = jest.fn()
+}
+class SelectionServiceMock {
+  selectRecords = jest.fn()
+  deselectRecords = jest.fn()
+  clearSelection = jest.fn()
+  selectedRecordsIdentifiers$ = new BehaviorSubject([])
+}
+
+describe('ResultsTableComponent', () => {
```

Comment:
I' don't see any css selector in these tests, even though there is rendering logic to tests (several `*ngIf`s). This would be easier to test if the component was split into a rendering part and a logical part (see my comment about a provided service). This is just a suggestion, I don't know the project well enough, might be tested in another way.

Reply:
You're right, several logic branches are not tested yet. This should be handled later when/if this component can be split up into logic/presentation further.